### PR TITLE
CNI to support transparent mode

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
This PR adds support for azure cni to support calico policies in Azure VMs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: network/endpoint_linux.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ const (
 	// Prefix for host virtual network interface names.
 	hostVEthInterfacePrefix = commonInterfacePrefix + "v"
 
+	transPrefix = "cali"
```
Comment (sharmasushant) [inline network/endpoint_linux.go]: Add a commen tto indicate this is prefix in transparent mode

File: network/endpoint_linux.go [hunk 2]
```diff
@@ -238,15 +251,18 @@ func addRoutes(interfaceName string, routes []RouteInfo) error {
 		nlRoute := &netlink.Route{
 			Family:    netlink.GetIpAddressFamily(route.Gw),
 			Dst:       &route.Dst,
+			Src:       route.Src,
```
Comment (sharmasushant) [inline network/endpoint_linux.go]: lets ensure that multitenancy works with this change.

File: network/endpoint_linux.go [hunk 3]
```diff
@@ -271,8 +287,11 @@ func deleteRoutes(interfaceName string, routes []RouteInfo) error {
...
```
Comment (sharmasushant) [inline network/endpoint_linux.go]: ensure that we can delete successfully for multitenancy

File: network/endpoint_linux.go (hunk 2 above)
Comment (tamilmani1989) [reply network/endpoint_linux.go]: we dint change anything here.. i can remove this also.  we are not setting source anywhere

File: network/endpoint_linux.go (hunk 3 above)
Comment (tamilmani1989) [reply network/endpoint_linux.go]: same as above

**Pull request merged**